ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh

```diff
@@ -14,11 +14,34 @@ cluster_name=${NAMESPACE}-${UNIQUE_HASH}
 
 echo "$(date -u --rfc-3339=seconds) - Setting up external load balancer"
 
-third_octet=$(grep -oP '[ci|qe\-discon]-segment-\K[[:digit:]]+' <(echo "${LEASED_RESOURCE}"))
+if [[ ${LEASED_RESOURCE} == *"vlan"* ]]; then
+  vlanid=$(grep -oP '[ci|qe\-discon]-vlan-\K[[:digit:]]+' <(echo "${LEASED_RESOURCE}"))
+
+  # ** NOTE: The first two addresses are not for use. [0] is the network, [1] is the gateway
+
+  dns_server=$(jq -r --arg VLANID "$vlanid" '.[$VLANID].dnsServer' /var/run/vault/vsphere-config/subnets.json)
+  gateway=$(jq -r --arg VLANID "$vlanid" '.[$VLANID].gateway' /var/run/vault/vsphere-config/subnets.json)
+  mask=$(jq -r --arg VLANID "$vlanid" '.[$VLANID].mask' /var/run/vault/vsphere-config/subnets.json)
+
+  external_lb_ip_address=$(jq -r --arg VLANID "$vlanid" '.[$VLANID].ipAddresses[2]' /var/run/vault/vsphere-config/subnets.json)
+
+  jq -r --argjson N 2 --arg VLANID "$vlanid" '.[$VLANID].ipAddresses[$N]' /var/run/vault/vsphere-config/subnets.json >> "${SHARED_DIR}"/vips.txt
+  jq -r --argjson N 3 --arg VLANID "$vlanid" '.[$VLANID].ipAddresses[$N]' /var/run/vault/vsphere-config/subnets.json >> "${SHARED_DIR}"/vips.txt
+  jq -r --arg VLANID "$vlanid" '.[$VLANID].machineNetworkCidr' /var/run/vault/vsphere-config/subnets.json >>  "${SHARED_DIR}"/machinecidr.txt
```

Comment:
should we check for missing VLANs in the Vault config? 
replace /var/run/vault/vsphere-config/subnets.json with a  var?